pkg/kube/secrets.go

```diff
@@ -0,0 +1,83 @@
+package kube
+
+import (
+	"context"
+
+	corev1 "github.com/rancher/wrangler-api/pkg/generated/controllers/core/v1"
+
+	"github.com/juju/errors"
+	k8scorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/cache"
+)
+
+type Secrets struct {
+	ctx                  context.Context
+	secretCache          corev1.SecretCache
+	reviewResultTTLCache *cache.LRUExpireCache
+}
+
+func NewSecrets(ctx context.Context, secretCache corev1.SecretCache) *Secrets {
+	return &Secrets{
+		ctx:                  ctx,
+		secretCache:          secretCache,
+		reviewResultTTLCache: cache.NewLRUExpireCache(1024),
```

Comment:
Why do we need to have LRU cache here? I think we are using `secretCache` so we have all sa token cached in memory?

Reply:
updated